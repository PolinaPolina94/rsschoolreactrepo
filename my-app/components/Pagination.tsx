import React, { useState } from 'react';
import styles from '../styles/Pagination.module.css';
import Link from 'next/link';
import { useRouter } from 'next/router';

const Pagination = () => {
  const router = useRouter();
  const name = router.query.name;
  const [totalItems] = useState(826);
  const [page, setPage] = useState(1);

  const pagesCount = Math.ceil(totalItems / 20);
  const pages = [];
  for (let i = 1; i <= pagesCount; i++) {
    pages.push(i);
  }
  const curP = page;
  const curPF = curP - 5 < 0 ? 0 : curP - 5;
  const curPL = curP + 5;
  const slicedPages = pages.slice(curPF, curPL);
  slicedPages.length <= 9;
  return (
    <div className={styles.container}>
      {slicedPages.map((el) => (
        <Link
          href={`/?page=${el}&name=${name}`}
          className={styles.a}
          // className={({ isActive }) => (isActive ? styles.active : undefined)}
          key={el + 1}
          onClick={() => {
            // e.preventDefault();
            // console.log(e.currentTarget.innerText);
            setPage(el);
          }}
        >
          {' '}
          {el}
        </Link>
      ))}
      <div data-testid="PaginatorContent"></div>
    </div>
  );
};
export default Pagination;
