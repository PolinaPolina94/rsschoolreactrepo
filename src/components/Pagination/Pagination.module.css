.container {
    display: flex;
    justify-content: space-between;
    max-width: 500px;
    margin: 0 auto;
    margin-top: 15px;
    gap: 10px;
    align-items: center;
    text-content: center;
}

.container.disabled {
    pointer-events: none;
}

a {
    border: 2px solid #ffd4ac;
    width:20px;
    height:20px;
    border-radius: 50%;
    text-align: center;
    background: #fffc8c;
    color: brown;
    font-weight: bold;
}

.item:hover {
    opacity: 0.4;
    cursor: pointer;
}

.item:active {
    opacity: 0.4;
}

.button {
    background: none;
    border: none;
    cursor: pointer;
    width: 30px;
    height: 30px
}
.button:hover {
    border-bottom: 2px solid white;
}

a {
    text-decoration: none;
}

a.active {
    opacity: 0.4
}