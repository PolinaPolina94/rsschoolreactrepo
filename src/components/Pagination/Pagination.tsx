import React, { useState } from 'react';
import classes from './Pagination.module.css';
import { NavLink } from 'react-router-dom';

export function Paginator() {
  const [totalItems] = useState(826);
  const [totalPages] = useState(42);
  const [page, setPage] = useState(1);
  console.log(totalPages);
  const activeStyle = localStorage.getItem('active');

  // useEffect(() => {
  //   fetch('https://rickandmortyapi.com/api/character')
  //     .then((res) => res.json())
  //     .then((info) => {
  //       setTotalItems(info.info.count);
  //       setTotalPages(info.info.pages);
  //     });
  // });

  const pagesCount = Math.ceil(totalItems / 20);
  const pages = [];
  for (let i = 1; i <= pagesCount; i++) {
    pages.push(i);
  }
  const curP = page;
  const curPF = curP - 5 < 0 ? 0 : curP - 5;
  const curPL = curP + 5;
  const slicedPages = pages.slice(curPF, curPL);
  slicedPages.length <= 9;
  return (
    <div
      className={
        activeStyle === 'rockNew'
          ? `${classes.container} ${classes.disabled}`
          : `${classes.container}`
      }
    >
      {slicedPages.map((el) => (
        <NavLink
          to={`/page/${el}`}
          className={({ isActive }) => (isActive ? classes.active : undefined)}
          key={el + 1}
          onClick={() => setPage(el)}
        >
          {' '}
          {el}
        </NavLink>
      ))}
      <div data-testid="PaginatorContent"></div>
    </div>
  );
}

// export default Paginator;
