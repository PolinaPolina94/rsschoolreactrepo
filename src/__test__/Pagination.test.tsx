import { fireEvent, render, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Paginator } from '../components/Pagination/Pagination';
// import { getPages } from './__ mocks __/usersApi';
import { BrowserRouter } from 'react-router-dom';

describe('Pagination test', () => {
  it('URL is updating when page is changing', async () => {
    render(
      <BrowserRouter>
        <Paginator />
      </BrowserRouter>
    );
    expect(screen.getByText('2')).toBeInTheDocument();
    await waitFor(() => {
      fireEvent.click(screen.getByText('2'));
      expect(window.location.search).toBe('');
    });
  });
});
