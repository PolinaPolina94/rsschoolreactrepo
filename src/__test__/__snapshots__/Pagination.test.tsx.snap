// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination test URL is updating when page is changing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <a
          href="/page/1"
        >
           
          1
        </a>
        <a
          href="/page/2"
        >
           
          2
        </a>
        <a
          href="/page/3"
        >
           
          3
        </a>
        <a
          href="/page/4"
        >
           
          4
        </a>
        <a
          href="/page/5"
        >
           
          5
        </a>
        <a
          href="/page/6"
        >
           
          6
        </a>
        <div
          data-testid="PaginatorContent"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <a
        href="/page/1"
      >
         
        1
      </a>
      <a
        href="/page/2"
      >
         
        2
      </a>
      <a
        href="/page/3"
      >
         
        3
      </a>
      <a
        href="/page/4"
      >
         
        4
      </a>
      <a
        href="/page/5"
      >
         
        5
      </a>
      <a
        href="/page/6"
      >
         
        6
      </a>
      <div
        data-testid="PaginatorContent"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
